ssvsigner/ekm/remote_key_manager.go

```diff
@@ -119,10 +141,31 @@ func (km *RemoteKeyManager) AddShare(
 // its highest attestation/proposal data locally. If the remote or local operations
 // fail, returns an error.
 func (km *RemoteKeyManager) RemoveShare(ctx context.Context, pubKey phase0.BLSPubKey) error {
-	if err := km.signerClient.RemoveValidators(ctx, pubKey); err != nil {
+	statuses, err := km.signerClient.RemoveValidators(ctx, pubKey)
+	if err != nil {
 		return fmt.Errorf("remove validator: %w", err)
 	}
 
+	for _, status := range statuses {
+		switch status {
+		case web3signer.StatusDeleted:
+			continue
+
+		case web3signer.StatusDuplicated:
+			// A failed request does not guarantee that the keys were not deleted.
+			// It's possible that the ssv-signer successfully deleted the keys,
+			// but a network error occurred before a response could be received.
+			// Or, if ssv-signer is behind a load balancer, the load balancer may return an error.
+			// In such cases, the node would crash and, upon restarting, encounter a not found key error.
```

Comment:
The status handling logic seems inconsistent. For `RemoveValidators`, we should expect `StatusNotFound` when trying to delete a non-existent key, not `StatusDuplicated`.

Consider changing:
```go
case web3signer.StatusDuplicated:
    // Comment mentions "not found key error" but checks for duplicated
```

to:
```go
case web3signer.StatusNotFound:
    // A failed request does not guarantee that the keys were not deleted...
```
The current logic doesn't align with the comment or the expected Web3Signer behavior.